Add pull mode to remote job execution (#1630)
theforeman/foreman-documentation: guides/common/modules/proc_configuring-a-host-to-use-the-pull-client.adoc

Managed hosts do not require an SSH connection to their {SmartProxyServer}.

.Prerequisites
* You have registered the host to {Project}.
* The host's {SmartProxy} is configured to use `pull-mqtt` mode.
* The {Team} {project-client-name} repository is enabled and synchronized on {ProjectServer}, and enabled on the host.
* The host is able to communicate with its {SmartProxy} over MQTT using port `1883`.
* The host is able to communicate with its {SmartProxy} over HTTPS.

NOTE: The `katello-pull-transport-migrate` package was created to help users migrate from `katello-agent` to remote execution with the pull client.
However, having `katello-agent` installed on the host is not a requirement.
You can use `katello-pull-transport-migrate` regardless of whether `katello-agent` is installed.

.Procedure
. Install the `katello-pull-transport-migrate` package on your host:

> ** On EL8 and EL9 hosts:

Reviewer: ```suggestion
** On {EL} 9 and {EL} 8 hosts:
```

+
----
# dnf install katello-pull-transport-migrate
----

> ** On EL7 hosts:

Reviewer: ```suggestion
** On {EL} 7 hosts:
```

+
----
# dnf install katello-pull-transport-migrate

> ----

Reviewer: I believe hosts should be mentioned in reversed order here: first EL 9 and EL 8 hosts section, then EL 7 hosts section.

+
The package installs `foreman_ygg_worker` and `yggdrasil` as dependencies and enables the pull mode on the host.
The host's `subscription-manager` configuration and consumer certificates are used to configure the `yggdrasil` client on the host, and the pull mode client worker is started.

> . Ensure the `yggdrasild` service runs on the host to execute remote jobs using the pull client:

Reviewer 2: This step is optional.
```suggestion
. Optional: To verify that the pull client is running and configured properly, check the status of the `yggdrasild` service:
```
